src/migration/MigratiorStatus.hpp

```diff
@@ -0,0 +1,91 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2022-2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+namespace migration {
+
+/**
+ * @brief The status of a migrator, it provides the helper functions to convert the status to string and vice versa
+ */
+class MigratorStatus {
+public:
+    /**
+     * @brief The status of a migrator
+     */
+    enum Status { Migrated, NotMigrated, NotKnown, Count };
```

Comment:
Probably should be `enum class`

Reply:
I compared the scoped enum and old enum. I want to use refer the status like `MigratorStatus::Migrated`, Instead of `MigratorStatus::Status::Migrated`. Is it too verbose ?

Reply:
I compared the scoped enum and old enum. I want to use refer the status like `MigratorStatus::Migrated`, Instead of `MigratorStatus::Status::Migrated`. Is it too verbose ?

Reply:
I also just now noticed the name of the file is messed up too 🔢